### PR TITLE
newsboat: Add a new module for newsboat

### PR DESCRIPTION
**Pull request by MasseR**
A program module for configuring newsboat. Or at least the initial module for it.

It supports right now just a few configuration values, stuff that I mostly use, as well as support for urls, tags and queries.

### CODE REVIEW

File: modules/programs/newsboat.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+let
+  cfg = config.programs.newsboat;
+  wrapQuote = x: "\"${x}\"";
+
+in
+
+{
+  options = {
+    programs.newsboat = {
+      enable = mkEnableOption "Enable newsboat";
```
[the remark this answers is not in the document]
Comment (MasseR) [reply modules/programs/newsboat.nix]: Oh this was news to me. Thanks

File: modules/programs/newsboat.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        description = "Whether to enable automatic reloading while newsboat is running";
+      };
+      reloadTime = mkOption {
+        type = types.nullOr types.int;
+        default = 60;
+        description = "Time between reloads";
+      };
+      browser = mkOption {
+        type = types.string;
+        default = "${pkgs.elinks}/bin/elinks";
```
[the remark this answers is not in the document]
Comment (MasseR) [reply modules/programs/newsboat.nix]: A good recommendation, thanks.

File: modules/programs/newsboat.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.newsboat ];
+    home.file.".newsboat/urls".text = ''
+      ${builtins.concatStringsSep "\n" (builtins.map (u: builtins.concatStringsSep " " ([u.url] ++ (builtins.map wrapQuote u.tags))) cfg.urls)}
+
+      ${builtins.concatStringsSep "\n" (builtins.map (q: "\"query:${q}:${builtins.replaceStrings ["\""] ["\\\""] cfg.queries."${q}"}\"") (builtins.attrNames cfg.queries))}
```
[the remark this answers is not in the document]
Comment (MasseR) [reply modules/programs/newsboat.nix]: Thanks for the builtins tip and especially for the `escape` shorthand!

**Comment by rycee** [comment]
Much nicer, thanks for the improvements :-)

Rebased to master in 4caa45b8bb2ee8aafcfaa2e96328127b9d19ce0f.

**Pull request closed without merging**